internal/playwright/config.go

```diff
@@ -225,6 +227,17 @@ func shardInSuites(rootDir string, suites []Suite, ccy int) ([]Suite, error) {
 
 		testFiles := fpath.ExcludeFiles(files, excludedFiles)
 
+		if s.ShardGrepEnabled && (s.Params.Grep != "" || s.Params.GrepInvert != "") {
+			var unmatched []string
+			testFiles, unmatched = grep.MatchFiles(os.DirFS(rootDir), testFiles, s.Params.Grep, s.Params.GrepInvert)
+			if len(testFiles) == 0 {
+				log.Warn().Str("suiteName", s.Name).Str("grep", s.Params.Grep).Msg("No files match the configured grep expressions")
+				continue
+			} else if len(unmatched) > 0 {
+				log.Info().Str("suiteName", s.Name).Str("grep", s.Params.Grep).Msgf("Files filtered out by grep: %q", unmatched)
```

Comment:
Saw something interesting that may not be directly related to your changes.

I ran with the following setup from our example repo:
```
suites:
  - name: "Firefox Win"
    platformName: "Windows 11"
    screenResolution: "1440x900"
    testMatch: ['tests/.*.js']
    shard: spec
    shardGrepEnabled: true
    params:
      grep: "@fast"
      browserName: "firefox"
      project: "firefox"
```

and saucectl logged:
```
15:27:05 INF Files filtered out by grep: ["node_modules/events/tests/add-listeners.js" "node_modules/events/tests/check-listener-leaks.js" "node_modules/events/tests/common.js" "node_modules/events/tests/errors.js" "node_modules/events/tests/events-list.js" "node_modules/events/tests/events-once.js" "node_modules/events/tests/index.js" "node_modules/events/tests/legacy-compat.js" "node_modules/events/tests/listener-count.js" "node_modules/events/tests/listeners-side-effects.js" "node_modules/events/tests/listeners.js" "node_modules/events/tests/max-listeners.js" "node_modules/events/tests/method-names.js" "node_modules/events/tests/modify-in-emit.js" "node_modules/events/tests/num-args.js" "node_modules/events/tests/once.js" "node_modules/events/tests/prepend.js" "node_modules/events/tests/remove-all-listeners.js" "node_modules/events/tests/remove-listeners.js" "node_modules/events/tests/set-max-listeners-side-effects.js" "node_modules/events/tests/special-event-names.js" "node_modules/events/tests/subclass.js" "node_modules/events/tests/symbols.js" "tests/demo-todo-app.spec.js"] grep=@fast suiteName="Firefox Win"
```

I was surprised to see that node_modules was even under consideration, especially because the .sauceignore file contains `node_modules/`.

Reply:
Yup that's not related, I'll open a ticket to address that in a separated PR (exclude any file matching the `.sauceignore` pattern.